### PR TITLE
Ks us 29 31

### PR DESCRIPTION
**Pull request by mikez321**
Krista's work.  Completes US 28-31.  All tests passing.  Potential issue when merging in story 32 since order statuses will become numbers, but Mike and Krista have discussed it so check with them for issues.

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -29,6 +28,13 @@ def create
     end
   end
 
+  def destroy
+    order = Order.find(params[:id])
+    order.cancel_order
```
Comment (mikez321) [inline app/controllers/orders_controller.rb:33]: Is this the whole "you don't actually destroy things, just don't show them" part of the readme?

File: app/models/item.rb [hunk 2]
```diff
@@ -41,4 +41,14 @@ def qty_purchased
     item_orders.sum(:quantity)
   end
 
+  def order_qty_purchased(order)
+  item_orders.where(order_id: order)
```
Comment (mikez321) [inline app/models/item.rb:45]: Getting to do some AR!  Nice!

File: config/routes.rb [hunk 3]
```diff
@@ -17,12 +17,12 @@
 
   resources :reviews, only: [:edit, :update, :destroy]
 
-  resources :orders, only: [:new, :create, :show]
+  resources :orders, only: [:new, :create, :index]
```
Comment (mikez321) [inline config/routes.rb:20]: Nice job updating these resources and not just adding!

File: db/migrate/20200411161613_add_order_default_status.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddOrderDefaultStatus < ActiveRecord::Migration[5.1]
```
Comment (mikez321) [inline db/migrate/20200411161613_add_order_default_status.rb:1]: I'm about to undo this, but nice job on the migrations!

File: db/schema.rb [hunk 5]
```diff
@@ -22,6 +22,7 @@
     t.integer "quantity"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
+    t.string "status"
```
Comment (mikez321) [inline db/schema.rb:25]: I wonder how these migrations work, I guess it just overwrites and then we have to re-merge anyways?

File: spec/features/orders/creation_spec.rb [hunk 6]
```diff
@@ -6,6 +6,9 @@
 # - Details of the order:
```
Comment (mikez321) [inline spec/features/orders/creation_spec.rb:6]: Probably want to remove these commented lines

File: spec/features/user_orders/show_spec.rb [hunk 7]
```diff
@@ -40,34 +40,54 @@
...
-        click_link "My Orders"
-        expect(current_path).to eq("/profile/orders")
-    end
+    it "I can visit the order show page from my order index page and see an individual orders details" do
+        visit "/profile/orders"
+        click_link "Order Number: #{@order1.id}"
+        expect(current_path).to  eq("/profile/orders/#{@order1.id}")
 
-    it "they can visit the order show page and see every order and all of each orders details" do
+        expect(page).to have_content("Order Number: #{@order1.id}")
```
Comment (mikez321) [inline spec/features/user_orders/show_spec.rb:48]: Ugh... these tests.... so long...

**Comment by Joshua-Tukman** [comment]
That is a bunch of great code Krista!

**Pull request merged**